### PR TITLE
Added libui

### PR DESCRIPTION
**Pull request by Frityet**
"*Simple and portable (but not inflexible) GUI library in C that uses the native GUI technologies of each platform it supports.*"

### CODE REVIEW

File: packages/l/libui/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    add_deps {
+        "meson", "ninja"
+    }
+
+    on_install(function (package)
+        import("package.tools.meson").install(package, {
+            "-Dexamples=false",
+            "-Dtests=false",
+            "--default-library=static",
+        })
```
Comment (waruqi) [inline packages/l/libui/xmake.lua:17]: add on_test

File: packages/l/libui/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package("libui")
+    set_homepage("https://libui-ng.github.io/libui-ng/")
+    set_description("A portable GUI library for C")
+
+    set_urls("https://github.com/libui-ng/libui-ng.git")
+    add_versions("3-12-22", "8c82e737eea2f8ab3667e227142abd2fd221f038")
+
+    add_deps {
+        "meson", "ninja"
+    }
```
Comment (waruqi) [inline packages/l/libui/xmake.lua:10]: add_deps("meson", "ninja")

File: packages/l/libui/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package("libui")
+    set_homepage("https://libui-ng.github.io/libui-ng/")
+    set_description("A portable GUI library for C")
+
+    set_urls("https://github.com/libui-ng/libui-ng.git")
+    add_versions("3-12-22", "8c82e737eea2f8ab3667e227142abd2fd221f038")
```
Comment (waruqi) [inline packages/l/libui/xmake.lua:6]: use date version, e.g. `2023.1.2`

File: packages/l/libui/xmake.lua [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+    add_deps {
+        "meson", "ninja"
+    }
+
+    on_install(function (package)
+        import("package.tools.meson").install(package, {
+            "-Dexamples=false",
+            "-Dtests=false",
+            "--default-library=static",
```
Comment (waruqi) [inline packages/l/libui/xmake.lua:16]: need handle shared/static library. `if package:config("shared") then`

**Comment by Frityet** [comment]
@waruqi fixed the specified issues issues

File: packages/l/libui/xmake.lua [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    on_install(function (package)
+        import("package.tools.meson").install(package, {
+            "-Dexamples=false",
+            "-Dtests=false",
+            "--default-library="..package:config("shared") and "shared" or "static",
+        })
+    end)
+
+    on_test(function (package)
+		assert(package:check_csnippets({test = [[
```
Comment (waruqi) [inline packages/l/libui/xmake.lua:19]: incorrect indent, spaces 4

File: packages/l/libui/xmake.lua [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+#include <stdio.h>
+#include <ui.h>
+
+int onClosing(uiWindow *w, void *data)
```
Comment (waruqi) [inline packages/l/libui/xmake.lua:23]: please use indent

File: packages/l/libui/xmake.lua [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    set_description("A portable GUI library for C")
+
+    set_urls("https://github.com/libui-ng/libui-ng.git")
+    add_versions("2022.12.3", "8c82e737eea2f8ab3667e227142abd2fd221f038")
+
+    add_deps("meson", "ninja")
+
+    on_install(function (package)
+        import("package.tools.meson").install(package, {
+            "-Dexamples=false",
```
Comment (waruqi) [inline packages/l/libui/xmake.lua:12]:
```lua
local configs = {-Dexamples=false, "-Dtests=false"}
table.insert(configs, "--default-library=" .. (package:config("shared") and "shared" or "static"))
import("package.tools.meson").install(package, configs)
```

File: packages/l/libui/xmake.lua [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+int main(void)
+{
+	uiInitOptions o = {0};
+	const char *err;
+	uiWindow *w;
+	uiLabel *l;
+
+	err = uiInit(&o);
+	if (err != NULL) {
+		fprintf(stderr, "Error initializing libui-ng: %s\n", err);
```
Comment (waruqi) [inline packages/l/libui/xmake.lua:38]: incorrent indent

File: packages/l/libui/xmake.lua [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+#include <ui.h>
+
+int onClosing(uiWindow *w, void *data)
+{
+	uiQuit();
+	return 1;
+}
+
+int main(void)
+{
```
Comment (waruqi) [inline packages/l/libui/xmake.lua:30]: do not use main
```
void test() {
```

**Review by Frityet — COMMENTED** [review]
Fixed

**Comment by Frityet** [comment]
@waruqi any updates on getting this merged?

**Comment by waruqi** [comment]
it need gtk+3 on linux, you can install it to test this package in linux ci. 

if it works, then we can remove it in ci.

**Comment by waruqi** [comment]
https://github.com/xmake-io/xmake-repo/pull/1772

**Pull request closed without merging**